src/Core/DataProvider/PDFDataProviderInterface.php

```diff
@@ -24,12 +24,17 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
-header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
-header('Last-Modified: '.gmdate('D, d M Y H:i:s').' GMT');
+namespace PrestaShop\PrestaShop\Core\DataProvider;
 
-header('Cache-Control: no-store, no-cache, must-revalidate');
-header('Cache-Control: post-check=0, pre-check=0', false);
-header('Pragma: no-cache');
-
-header('Location: ../../../../../../');
-exit;
+/**
+ * Interface PDFDataProviderInterface provides data regarding PDF
+ */
+interface PDFDataProviderInterface
+{
+    /**
+     * Gets invoice template type
+     *
+     * @return string
+     */
+    public function getInvoiceTemplateType();
```

Comment:
imo method name is too specific and interface name is too generic. @mickaelandrieu ?

maybe method could be renamed to `getTemplateType()` and interface to `PdfTemplateTypeProvider` so we may have `InvoicePdfTemplateTypeProvider implements PdfTemplateTypeProvider` and then another for `RefundPdfTemplateProvider implements PdfTemplateTypeProvider` if needed?

Reply:
good idea!